research/delf/delf/python/training/global_features_utils.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2021 The TensorFlow Authors All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utilities for the global model training."""
+
+import os
+
+from absl import flags
+from absl import logging
+
+import tensorflow as tf
+import numpy as np
+
+from tensorboard import program
+
+from delf.python.datasets.revisited_op import dataset
+
+FLAGS = flags.FLAGS
+
+
+class AverageMeter():
+  """Computes and stores the average and current value of loss."""
+
+  def __init__(self):
+    """Initialization of the AverageMeter."""
+    self.reset()
+
+  def reset(self):
+    """Resets all the values."""
+    self.val = 0
+    self.avg = 0
+    self.sum = 0
+    self.count = 0
+
+  def update(self, val, n=1):
+    """Updates values in the AverageMeter.
+
+    Args:
+      val: Float, loss value.
+      n: Integer, number of instances.
+    """
+    self.val = val
+    self.sum += val * n
+    self.count += n
+    self.avg = self.sum / self.count
+
+
+def compute_metrics_and_print(dataset_name, sorted_index_ids, ground_truth,
+                              desired_pr_ranks=[1, 5, 10], log=True):
+  """Computes and loggs easy/medium/hard ground-truth metrics for Revisited
```

Comment:
first line in docstring should not exceed 80 chars.
Ref: https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings